Clarify React18 concurrent root and remove opt-out for 0.74 (#4011)
facebook/react-native-website: docs/react-18-and-react-native.md

Previous versions of React Native built on the old architecture **cannot** support concurrent rendering or concurrent features. This is because the old architecture relied on mutating the native trees, which doesn’t allow for React to prepare multiple versions of the tree at the same time.

Fortunately, the New Architecture was written bottom-up with concurrent rendering in mind, and is fully compatible with React 18. This means, in order to upgrade to React 18 in your React Native app, your application **needs to use the React Native's New Architecture** including Fabric Native Components and Turbo Native Modules.

This means you’re able to use the new features in React 18 as soon as flip the New Architecture switch. Since the new concurrent features are opt-in by using features like `startTransition` or `Suspense`, we expect React 18 to work out-of-the-box with minimal changes for users who migrate to the New Architecture or create a new app with the New Architecture enabled.

> ### Users not yet migrated to the New Architecture

Reviewer: ```suggestion
### Note for the Old Architecture users
```

Alternate heading proposition.

> Users that are still on the Old Architecture, will still use React 17 mode even if React 18 is installed in the `package.json` file.

Reviewer: ```suggestion
Apps that are still on the Old Architecture will use React 17 mode even if React 18 is listed as a dependency in the `package.json` file.
```

Wording nit.
